### PR TITLE
discount_id issue

### PR DESCRIPTION
**Pull request by fivetran-jamie**
**Are you a current Fivetran customer?** 

fivetran made pr

**What change(s) does this PR introduce?** 

uses code instead of discount_id for granularity stuff (tests and row_number partitioning)

**Did you update the CHANGELOG?** 

- [x] Yes

**Does this PR introduce a breaking change?**

- [ ] Yes (please provide breaking change details below.)
- [x] No  (please provide an explanation as to how the change is non-breaking below.)

**Did you update the dbt_project.yml files with the version upgrade (please leverage standard semantic versioning)? (In both your main project and integration_tests)** 

- [x] Yes

**Is this PR in response to a previously created Bug or Feature Request**

- [x] Yes, Issue/Feature https://github.com/fivetran/dbt_shopify_source/issues/54
- [ ] No 

**How did you test the PR changes?** 

 

- [ ] Buildkite  
- [x] Local (please provide additional testing details below)

**Select which warehouse(s) were used to test the PR**
 

- [x] BigQuery
- [ ] Redshift
- [ ] Snowflake
- [ ] Postgres
- [ ] Databricks
- [ ] Other (provide details below)

**Provide an emoji that best describes your current mood**
 
🍰 

**Feedback**

We are so excited you decided to contribute to the Fivetran community dbt package! We continue to work to improve the packages and would greatly appreciate your [feedback](https://www.surveymonkey.com/r/DQ7K7WW) on our existing dbt packages or what you'd like to see next.

### CODE REVIEW

File: models/stg_shopify.yml [hunk 1]
```diff
@@ -1570,7 +1569,7 @@ models:
       - name: created_at
         description: When the checkout discount application was created.
       - name: discount_id
-        description: ID of the discount.
+        description: ID of the discount. Use `code` instead.
```
Comment (fivetran-catfritz) [inline models/stg_shopify.yml:1572]:
```suggestion
        description: ID of the discount. Deprecated, use `code` instead.
```

File: models/src_shopify.yml [hunk 2]
```diff
@@ -1340,7 +1340,7 @@ sources:
           - name: created_at
             description: When the checkout discount application was created.
           - name: discount_id
-            description: ID of the discount.
+            description: ID of the discount. Use `code` instead.
```
Comment (fivetran-catfritz) [inline models/src_shopify.yml:1343]:
```suggestion
            description: ID of the discount. Deprecated, use `code` instead.
```
Might be helpful to explain why we need to use code instead. You can rephrase of course!

**Review by fivetran-catfritz — APPROVED** [review]
One small suggestion for documentation, but approved pending customer review!

File: models/stg_shopify.yml (hunk 1 above)
Comment (fivetran-jamie) [reply models/stg_shopify.yml:1572]: yeah i originally had this, but I couldn't confirm in the shopify docs that this field is specifically _deprecated_. i only found unanswered questions about the field in the forums, so might wanna keep it as is?

Comment (fivetran-catfritz) [reply models/stg_shopify.yml:1572]: I did see it's deprecated in the [Fivertran release notes](https://fivetran.com/docs/applications/shopify/changelog#july2021), so maybe that's enough confirmation? If not, makes sense to leave it as-is!

**Pull request merged**